src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -609,7 +621,8 @@ private Task ProcessRstStreamFrameAsync()
             }
 
             ThrowIfIncomingFrameSentToIdleStream();
-            AbortStream(_incomingFrame.StreamId, new IOException(CoreStrings.Http2StreamResetByClient));
+            // No additional inbound header or data frames are allowed for this stream after receiving a reset.
```

Comment:
From the spec, it also seems like we should throw if we receive a second rst frame after receiving the first rst frame.

Reply:
Same for window update.

Reply:
Hmm, those are going to be tough to enforce. I can check for them while the stream is still active but we ignore them if they arrive after the stream is removed from the collection. The window updates could be handled as part of a cool down, but we'd only planned on tracking server aborted streams for that.

Reply:
Checking while the streams are active should be fine for now. These are supposed to be stream errors anyway which would reset the stream that's already no longer active.